test/e2e/delete_test.go

```diff
@@ -2,21 +2,90 @@ package e2e
 )
 
 func deletionTestSuite(t *testing.T) {
 	t.Run("Deletion", func(t *testing.T) { testWindowsNodeDeletion(t) })
 }
 
+// clearWindowsInstanceConfigMap removes all entries in the windows-instances configmap
+func (tc *testContext) clearWindowsInstanceConfigMap() error {
+	cm, err := tc.client.K8s.CoreV1().ConfigMaps(tc.namespace).Get(context.TODO(), controllers.InstanceConfigMap,
+		meta.GetOptions{})
+	if err != nil {
+		return errors.Wrap(err, "error retrieving windows-instances ConfigMap")
+	}
+	cm.Data = map[string]string{}
+	_, err = tc.client.K8s.CoreV1().ConfigMaps(tc.namespace).Update(context.TODO(), cm, meta.UpdateOptions{})
+	if err != nil {
+		return errors.Wrap(err, "error clearing windows-instances ConfigMap data")
+	}
+	return nil
+}
+
+// testBYOHRemoval tests that nodes are properly removed and deconfigured after removing configmap entries
+func (tc *testContext) testBYOHRemoval(t *testing.T) {
+	// Get list of BYOH nodes before the node objects are deleted
+	byohNodes, err := tc.listFullyConfiguredWindowsNodes(true)
+	require.NoError(t, err, "error getting BYOH node list")
+
+	// Remove all entries from the windows-instances ConfigMap, causing all node objects to be deleted
+	require.NoError(t, tc.clearWindowsInstanceConfigMap(),
+		"error removing windows-instances ConfigMap entries")
+	err = tc.waitForWindowsNodes(0, true, false, true)
+	require.NoError(t, err, "Removing ConfigMap entries did not cause Windows node deletion")
+
+	// For each node that was deleted, check that all the expected services have been removed
+	for _, node := range byohNodes {
+		t.Run(node.GetName(), func(t *testing.T) {
+			addr, err := tc.getAddress(node.Status.Addresses)
+			require.NoError(t, err, "unable to get node address")
+			svcs, err := tc.getWinServices(addr)
+			require.NoError(t, err, "error getting service map")
+			for _, svcName := range win.InstalledSVCs {
+				t.Run(svcName, func(t *testing.T) {
+					require.NotContains(t, svcs, svcName, "service present")
+				})
+			}
+			dirsCleanedUp, err := tc.requiredDirsDoNotExist(addr)
+			require.NoError(t, err, "error determining if created directories exist")
+			assert.True(t, dirsCleanedUp, "directories not removed")
+		})
+	}
+}
+
+// requiredDirsDoNotExist checks if the required directories created by WMCO exist on the instance with the provided address
+func (tc *testContext) requiredDirsDoNotExist(address string) (bool, error) {
+	command := "powershell.exe -NonInteractive -ExecutionPolicy Bypass -Command \""
+	for _, dir := range win.RequiredDirectories {
+		command += fmt.Sprintf("if ((Test-Path %s) -eq $true) { Write-Output %s exists}", dir, dir)
+	}
+	command += "exit 0\""
+	out, err := tc.runSSHJob("check-win-dirs", command, address)
+	if err != nil {
+		return false, errors.Wrapf(err, "error confirming directories do not exist %s", out)
+	}
+	if strings.Contains(out, "exists") {
+		return false, nil
+	}
+	return true, nil
```

Comment:
Can be simplified to `return !strings.Contains(out, "exists"), nil`

Overall though, the function is a bit strange to me as the positive case reads as a double negative: `false, the required dirs do not exist`
Maybe rename it to `requiredDirsExist`, and invert the return values and usage?

Reply:
I suggest calling it `checkDirsDoNotExist()`